python/pyiceberg/expressions/visitors.py

```diff
@@ -60,125 +60,127 @@
     PrimitiveType,
 )
 
+R = TypeVar("R")
```

Comment:
Since `T` is no longer imported, this could continue to use a local `T` typevar and we wouldn't need all the changes to this file.

Reply:
Yes, that's true, I've reverted the change